source/common/quic/envoy_quic_utils.h

```diff
@@ -45,16 +45,26 @@ class HeaderValidator {
   validateHeader(const std::string& header_name, absl::string_view header_value) = 0;
 };
 
+static constexpr absl::string_view invalid_underscore = "http3.unexpected_underscore";
+static constexpr absl::string_view too_many_http_headers = "http3.too_many_http_headers";
```

Comment:
Maybe add/reuse those already defined in Http3ResponseCodeDetailValues? You may need to move it around.